devito/finite_differences/finite_difference.py

```diff
@@ -284,4 +284,7 @@ def indices_weights_to_fd(expr, dim, inds, weights, matvec=1):
 
     deriv = EvalDerivative(*terms, base=expr)
 
+    # Evaluate remaining part of expression
+    deriv = deriv.evaluate
```

Comment:
AS I said this is dangerous because now you can prevent shifts cancelling out and by forcing the "avg" to be evaluated this can lead to quite worse accuracy. This evaluate should always be the most last part in case there is half-node indices left.

Reply:
there are two issues here:

1) there's no failing test in the test suite capturing the behavior you're describing. Could you add one? Pushing to this PR would be completely fine

2) 
> This evaluate should always be the most last part in case there is half-node indices left.

I'm not sure I see how my change would alter the order? The `return deriv` below would return control to `derivative.py._eval_fd`, so... what am I missing? perhaps with the MFE I'll see it

Reply:
my brain is not working very well these days, it doesn't change anything to have moved it.

Reply:
OK, no problem